src/sql/azdata.proposed.d.ts

```diff
@@ -403,6 +403,25 @@ declare module 'azdata' {
 		title: string;
 	}
 
+	export interface ConnectionProvider extends DataProvider {
+		/**
+		 * Changes the password for a connection profile (in case of the password expiration scenario)
+		 */
+		changePassword(connectionUri: string, connectionInfo: ConnectionInfo, newPassword: string): Thenable<PasswordChangeResult>;
```

Comment:
This should be optional - we can't guarantee that existing providers will implement it and this isn't something that currently needs to be required (especially since it's currently only for MSSQL...)

```suggestion
		changePassword?(connectionUri: string, connectionInfo: ConnectionInfo, newPassword: string): Thenable<PasswordChangeResult>;
```